pkg/controller/deployment/sync.go

```diff
@@ -549,3 +550,15 @@ func (dc *DeploymentController) isScalingEvent(ctx context.Context, d *apps.Depl
 	}
 	return false, nil
 }
+
+// generateReplicaSetName generates a ReplicaSet name by adding a pod template spec hash
+// to a Deployment name, optionally truncating it to ensure the ReplicaSet name is within the limit.
+func generateReplicaSetName(deploymentName, podTemplateSpecHash string) string {
+	maxDeploymentNameLength := validation.DNS1123SubdomainMaxLength - 1 - len(podTemplateSpecHash)
```

Comment:
nit: The number 1 (for the hyphen separator IIUC) should likely have a comment explaining what the 1 represents.

Reply:
I like the latter. Updated.